web/src/lib/components/pages/record.tsx

```diff
@@ -174,22 +172,20 @@ export default class RecordPage extends Component<RecordProps, RecordState> {
       uploading: true
     });
 
-    this.props.onSubmit(this.state.recordings, this.state.sentences, this.onProgress)
-      .then(() => {
-        this.reset();
-        this.tracker.trackSubmitRecordings();
-      })
-      .catch(() => {
-        this.setState({
-          uploading: false
-        });
-        confirm('You did not agree to our Terms of Service. Do you want to delete your recordings?', 'Keep the recordings', 'Delete my recordings').then((keep) => {
-          if (!keep) {
-            this.reset();
-            this.props.navigate('/');
-          }
-        })
+    try {
+      await this.props.onSubmit(this.state.recordings, this.state.sentences, this.onProgress);
+      this.reset();
+      this.tracker.trackSubmitRecordings();
+    } catch (e) {
+      this.setState({
+        uploading: false
       });
+      const keep = await confirm('You did not agree to our Terms of Service. Do you want to delete your recordings?', 'Keep the recordings', 'Delete my recordings')
```

Comment:
This catch can now happen if a single upload request failed (as well as not agreeing to privacy terms). Let's change this message to something like `Upload aborted. Do you want to delete your recordings?`